File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/metrics/PhoneState.java
package com.mapbox.services.android.navigation.v5.navigation.metrics;

import android.content.Context;

import com.mapbox.android.telemetry.TelemetryUtils;

/**
 * Class that will hold the current states of the device phone.
 */
public class PhoneState {
  private final int volumeLevel;
  private final int batteryLevel;
  private final int screenBrightness;
  private final boolean batteryPluggedIn;
  private final String connectivity;
  private final String audioType;
  private final String applicationState;
  private final String created;
  private final String feedbackId;
  private final String userId;

  public PhoneState(Context context) {
    this.volumeLevel = NavigationUtils.obtainVolumeLevel(context);
    this.batteryLevel = TelemetryUtils.obtainBatteryLevel(context);
    this.screenBrightness = NavigationUtils.obtainScreenBrightness(context);
    this.batteryPluggedIn = TelemetryUtils.isPluggedIn(context);
    this.connectivity = TelemetryUtils.obtainCellularNetworkType(context);
    this.audioType = NavigationUtils.obtainAudioType(context);
    this.applicationState = TelemetryUtils.obtainApplicationState(context);
    this.created = TelemetryUtils.obtainCurrentDate();
    this.feedbackId = TelemetryUtils.obtainUniversalUniqueIdentifier();
    this.userId = TelemetryUtils.retrieveVendorId();
  }

  String getUserId() {
    return userId;
  }

  int getVolumeLevel() {
    return volumeLevel;
  }

  int getBatteryLevel() {
    return batteryLevel;
  }

  int getScreenBrightness() {
    return screenBrightness;
  }

  boolean isBatteryPluggedIn() {
    return batteryPluggedIn;
  }

  String getConnectivity() {
    return connectivity;
  }

  String getAudioType() {
    return audioType;
  }

  String getApplicationState() {
    return applicationState;
  }

  String getCreated() {
    return created;
  }

  String getFeedbackId() {
    return feedbackId;
  }
}

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/metrics/PhoneState.kt
package com.mapbox.services.android.navigation.v5.internal.navigation.metrics

import android.content.Context
import com.mapbox.android.telemetry.TelemetryUtils

/**
 * Class that will hold the current states of the device phone.
 */
data class PhoneState(val context: Context) {
    val volumeLevel: Int = NavigationUtils.obtainVolumeLevel(context)
    val batteryLevel: Int = TelemetryUtils.obtainBatteryLevel(context)
    val screenBrightness: Int = NavigationUtils.obtainScreenBrightness(context)
    val isBatteryPluggedIn: Boolean = TelemetryUtils.isPluggedIn(context)
    val connectivity: String? = TelemetryUtils.obtainCellularNetworkType(context)
    val audioType: String = NavigationUtils.obtainAudioType(context)
    val applicationState: String = TelemetryUtils.obtainApplicationState(context)
    val created: String = TelemetryUtils.obtainCurrentDate()
    val feedbackId: String = TelemetryUtils.obtainUniversalUniqueIdentifier()
    val userId: String = TelemetryUtils.retrieveVendorId()
}
